### DISCUSSION TITLE
Uvicorn worker processes die on startup when `PYTHONDONTWRITEBYTECODE=1`

### DISCUSSION THREAD

**Discussion by winstxnhdw** [Kludex/uvicorn#2396] [Potential Issue] [answered]
To reproduce:

```bash
export PYTHONDONTWRITEBYTECODE=1
uvicorn main:app --workers 2
```

I have tested this on Debian Bookworm. Might want to make sure you do not have any `*.pyc` files as well.

**Comment by Kludex** [comment]
@abersheeran Do you know something about this?

**Reply by abersheeran** [reply]
This problem has never occurred in our environment. We have always run uvicorn/zibai in the docker container with PYTHONDONTWRITEBYTECODE=1.

**Reply by winstxnhdw** [reply]
It’s possible that you aren’t docker ignoring your *.pyc file which is copied over into the container.

**Comment by abersheeran** [comment]
Can you provide more information? For example, Traceback

**Reply by winstxnhdw** [reply]
I would provide a traceback if there was any but all I get is “Child process has died.” I’ll create a minimal example in a bit.

**Reply by abersheeran** [reply]
I recommend checking your code first to see if there is anything that would cause the process to exit. As far as I know, Uvicorn does not check pyc

**Reply by winstxnhdw** [reply, marked as answer]
False alarm. I had `pyannote.audio` imported in one of the script that I forgot to comment out. This causes Uvicorn workers to die for whatever reason. Why this only happens when `workers` are set to `2` and `PYTHONDONTWRITEBYTECODE=1` is a mystery, but at least it's not Uvicorn's problem.

**Reply by winstxnhdw** [reply]
FWIW, I just migrated over to Granian with workers today, and it doesn't have this problem. It's likely that Uvicorn is hitting some edge case here.
